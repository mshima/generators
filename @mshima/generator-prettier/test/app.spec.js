import path, { dirname } from 'path';
import expect from 'expect';
import helpers from 'yeoman-test';

import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));

describe('@mshima/prettier:app', () => {
  let runContext;
  let runResult;

  beforeEach(async () => {
    runContext = helpers.create('@mshima/prettier:app').withLookups([{ npmPaths: path.join(__dirname, '..', '..') }]);
  });

  afterEach(() => {
    runContext.cleanTestDirectory();
  });

  describe('with sample values', () => {
    beforeEach(async () => {
      runResult = await runContext.run();
    });

    it('should match snapshot', () => {
      expect(runResult.getSnapshot()).toMatchSnapshot();
    });
  });
});
