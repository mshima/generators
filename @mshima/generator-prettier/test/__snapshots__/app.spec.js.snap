// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mshima/prettier:app with sample values should match snapshot 1`] = `
Object {
  ".prettierrc.json": Object {
    "contents": "{
  \\"singleQuote\\": true,
  \\"printWidth\\": 120,
  \\"arrowParens\\": \\"avoid\\"
}
",
    "stateCleared": "modified",
  },
  "package.json": Object {
    "contents": "{
  \\"scripts\\": {
    \\"pretest\\": \\"concurrently -m 1 npm:pretest:*\\",
    \\"pretest:prettier\\": \\"npm run prettier:check\\",
    \\"fix\\": \\"concurrently -m 1 npm:fix:*\\",
    \\"fix:prettier\\": \\"npm run prettier:write\\",
    \\"prettier:write\\": \\"prettier --write .\\",
    \\"prettier:check\\": \\"prettier --check .\\"
  },
  \\"devDependencies\\": {
    \\"concurrently\\": \\"^4.1.0\\",
    \\"prettier\\": \\"2.3.0\\"
  }
}
",
    "stateCleared": "modified",
  },
}
`;
